### PR TITLE
LibGfx/PortableFormat: Easy optimisations

### PR DESCRIPTION
**Pull request by LucasChollet**
These two patches already reduce execution time by more than 30%.

Many low-hanging fruits remain tho

### CODE REVIEW

File: Userland/Libraries/LibGfx/ImageFormats/PBMLoader.cpp [hunk 1]
```diff
@@ -12,19 +12,19 @@ namespace Gfx {
...
 
     auto const context_size = context.width * context.height;
-    color_data.resize(context_size);
 
     if (context.type == PBMLoadingContext::Type::ASCII) {
         for (u64 i = 0; i < context_size; ++i) {
             auto const byte = TRY(stream.read_value<u8>());
             if (byte == '0')
-                color_data[i] = Color::White;
+                context.bitmap->set_pixel(i % context.width, i / context.width, Color::White);
```
Comment (nico) [inline Userland/Libraries/LibGfx/ImageFormats/PBMLoader.cpp:25]: You could also do `bitmap->begin()[i] = Color::White.value()` (doesn't bounds check then though)

**Review by nico — APPROVED** [review]
Seems alright.

File: Userland/Libraries/LibGfx/ImageFormats/PBMLoader.cpp (hunk 1 above)
Comment (LucasChollet) [reply Userland/Libraries/LibGfx/ImageFormats/PBMLoader.cpp:25]: I'm not a fan as it also doesn't check for the storage type. I know that it shouldn't be an issue here, but let's keep it the safe way until we have something better

File: AK/MemoryStream.cpp [hunk 2]
```diff
@@ -54,6 +54,17 @@ ErrorOr<Bytes> FixedMemoryStream::read_some(Bytes bytes)
     return bytes.trim(to_read);
 }
 
+ErrorOr<void> FixedMemoryStream::read_until_filled(AK::Bytes bytes)
+{
+    if (remaining() < bytes.size())
+        return Error::from_string_view_or_print_error_and_return_errno("Can't read past the end of the stream memory"sv, EINVAL);
```
Comment (kleinesfilmroellchen) [inline AK/MemoryStream.cpp:60]: This is not an error exposed from the Kernel, so you should just use `from_string_view` or `EINVAL`.

**Review by gmta — APPROVED** [review]

File: AK/MemoryStream.cpp (hunk 2 above)
Comment (timschumi) [reply AK/MemoryStream.cpp:60]: > This is not an error exposed from the Kernel

Are you sure? There is no reason why it couldn't be.

**Pull request merged**